File: EXAMEN VIERNES/sistema-reserva-libros/src/main/java/com/diegoAlvarez/sistema_reserva_libros/service/ReservaService.java
package com.diegoAlvarez.sistema_reserva_libros.service;

import com.diegoAlvarez.sistema_reserva_libros.dto.ReservaDTO;
import com.diegoAlvarez.sistema_reserva_libros.model.Reserva;

import java.util.List;

public interface ReservaService {
    Reserva crearReserva(ReservaDTO reservaDTO);
    List<Reserva> obtenerReservasPorUsuario(Long usuarioId);
}

File: EXAMEN VIERNES/FRONTEND/sistema-reserva-libros-frontend/src/app/services/reserva.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/api-response.model';
import { Reserva, ReservaDTO } from '../models/reserva.model';

@Injectable({
  providedIn: 'root'
})
export class ReservaService {
  private readonly url = 'http://localhost:8080/api/reservas';

  constructor(private readonly http: HttpClient) { }

  crearReserva(reserva: ReservaDTO): Observable<ApiResponse<Reserva>> {
    return this.http.post<ApiResponse<Reserva>>(this.url, reserva);
  }

  obtenerReservasPorUsuario(usuarioId: number): Observable<ApiResponse<Reserva[]>> {
    return this.http.get<ApiResponse<Reserva[]>>(`${this.url}/usuario/${usuarioId}`);
  }
}